File: shared/scripts/sync.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");

const ROOT = path.join(__dirname, "..", "..");
const sharedContent = path.join(ROOT, "shared", "content");
const sharedAssets = path.join(ROOT, "shared", "assets");
const androidAssets = path.join(ROOT, "android", "app", "src", "main", "assets");

function copyDir(src, dest) {
  if (!fs.existsSync(src)) return;
  fs.mkdirSync(dest, { recursive: true });
  for (const ent of fs.readdirSync(src, { withFileTypes: true })) {
    const s = path.join(src, ent.name);
    const d = path.join(dest, ent.name);
    if (ent.isDirectory()) copyDir(s, d);
    else fs.copyFileSync(s, d);
  }
}

fs.mkdirSync(androidAssets, { recursive: true });
if (fs.existsSync(sharedContent)) copyDir(sharedContent, path.join(androidAssets, "content"));
if (fs.existsSync(sharedAssets)) copyDir(sharedAssets, androidAssets);
console.log("Synced shared -> android assets");

File: shared/scripts/sync.py
#!/usr/bin/env python3
"""Copy shared/content and shared/assets into ios and android bundles."""

from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent.parent
sys.path.insert(0, str(ROOT.parent / ".cursor" / "skills" / "_shared"))
if not (ROOT.parent / ".cursor" / "skills" / "_shared").exists():
    sys.path.insert(0, str(Path.home() / ".cursor" / "skills" / "_shared"))
from monorepo_util import is_monorepo, sync_shared_to_platforms  # noqa: E402

if __name__ == "__main__":
    root = ROOT
    if not is_monorepo(root):
        print("Error: run from monorepo root (needs shared/config/app.json)", file=sys.stderr)
        sys.exit(1)
    sync_shared_to_platforms(root)
    print(f"Synced shared -> ios + android ({root})")
